ui/components/app/currency-input/currency-input.js

```diff
@@ -41,84 +45,93 @@ export default function CurrencyInput({
   onPreferenceToggle,
   swapIcon,
   className = '',
+  // if null, the asset is the native currency
+  asset,
 }) {
   const t = useContext(I18nContext);
 
+  const assetDecimals = asset?.decimals || NATIVE_CURRENCY_DECIMALS;
+
   const preferredCurrency = useSelector(getNativeCurrency);
   const secondaryCurrency = useSelector(getCurrentCurrency);
-  const conversionRate = useSelector(getConversionRate);
+
+  const primarySuffix =
+    asset?.symbol || preferredCurrency || EtherDenomination.ETH;
+  const secondarySuffix = secondaryCurrency.toUpperCase();
+  const isLongSymbol = (primarySuffix?.length || 0) > LARGE_SYMBOL_LENGTH;
+
+  const [shouldDisplayFiat, setShouldDisplayFiat] = useState(featureSecondary);
```

Comment:
What's the point of maintaining this state within the component when `swap` updates the `sendInputCurrencySwitched` state, which the initial value is based on? I'd remove this and just use `getIsFiatPrimary` directly

Reply:
It's to keep it agnostic from non-global (i.e., send-specific) state variables since it's more general based on its place in the directory (i.e., `ui/components` rather than `ui/components/multchain`